packages/fela-bindings/src/ThemeProviderFactory.js

```diff
@@ -20,7 +20,9 @@ export default function ThemeProviderFactory(
     }
 
     componentWillReceiveProps(nextProps: Object): void {
-      this.theme.update(nextProps.theme)
+      if (this.props.theme !== nextProps.theme) {
```

Comment:
Could you still use shallowEqual here? !== checking objects might cause false-positive values.